packages/debug/src/browser/view/debug-threads-source.tsx

```diff
@@ -44,7 +44,8 @@ export class DebugThreadsSource extends TreeSource {
         if (this.model.sessionCount === 1 && this.model.session && this.model.session.threadCount) {
             return this.model.session.threads;
         }
-        return this.model.sessions;
+        const sessions = [...this.model.sessions].filter(e => !e.parentSession);
+        return sessions.values();
```

Comment:
Since we're using iterators, why not do:

```ts
    *getElements(): IterableIterator<TreeElement> {
        if (this.model.sessionCount === 1 && this.model.session && this.model.session.threadCount) {
            return yield* this.model.session.threads;
        }
        for (const session of this.model.sessions) {
            if (session.parentSession) {
                yield session;
            }
        }
    }
```

`Array.values` seems to work but I was surprised to see it.

Reply:
Converted this method and another one in `debug-session.tsx` to use iterators.